src/formatters/mdDeployResultFormatter.ts

```diff
@@ -0,0 +1,251 @@
+
+      const tests = this.sortTestResults(failures);
+
+      this.ux.log('');
+      this.ux.styledHeader(chalk.red(`Test Failures [${this.result.response.details.runTestResult?.numFailures}]`));
+      this.ux.table(tests, {
+        columns: [
+          { key: 'name', label: 'Name' },
+          { key: 'methodName', label: 'Method' },
+          { key: 'message', label: 'Message' },
+          { key: 'stackTrace', label: 'Stacktrace' },
+        ],
+      });
+    }
+  }
+
+  protected verboseTestSuccess(): void {
+    const success = toArray(this.result?.response?.details?.runTestResult?.successes);
+    if (success.length) {
+      const tests = this.sortTestResults(success);
+      this.ux.log('');
+      this.ux.styledHeader(chalk.green(`Test Success [${success.length}]`));
+      this.ux.table(tests, {
+        columns: [
+          { key: 'name', label: 'Name' },
+          { key: 'methodName', label: 'Method' },
+        ],
+      });
+    }
+    const codeCoverage = toArray(this.result?.response?.details?.runTestResult?.codeCoverage);
+
+    if (codeCoverage.length) {
+      const coverage = codeCoverage.sort((a, b) => {
+        return a.name.toUpperCase() > b.name.toUpperCase() ? 1 : -1;
+      });
+
+      this.ux.log('');
+      this.ux.styledHeader(chalk.blue('Apex Code Coverage'));
+
+      coverage.map((cov: CodeCoverage & { lineNotCovered: string }) => {
+        const numLocationsNum = parseInt(cov.numLocations, 10);
+        const numLocationsNotCovered: number = parseInt(cov.numLocationsNotCovered, 10);
+        const color = numLocationsNotCovered > 0 ? chalk.red : chalk.green;
+
+        let pctCovered = 100;
+        const coverageDecimal: number = parseFloat(
+          ((numLocationsNum - numLocationsNotCovered) / numLocationsNum).toFixed(2)
+        );
+        if (numLocationsNum > 0) {
+          pctCovered = coverageDecimal * 100;
+        }
+        cov.numLocations = color(`${pctCovered}%`);
+
+        if (!cov.locationsNotCovered) {
+          cov.lineNotCovered = '';
+        }
+        const locations = toArray(cov.locationsNotCovered);
+        cov.lineNotCovered = locations.map((location) => location.line).join(',');
+      });
+
+      this.ux.table(coverage, {
+        columns: [
+          { key: 'name', label: 'Name' },
+          {
+            key: 'numLocations',
+            label: '% Covered',
+          },
+          {
+            key: 'lineNotCovered',
+            label: 'Uncovered Lines',
+          },
+        ],
+      });
+    }
+  }
+
+  protected verboseTestTime(): void {
+    if (
+      this.result.response?.details?.runTestResult?.successes ||
+      this.result?.response?.details?.runTestResult?.failures
+    ) {
+      this.ux.log('');
+      this.ux.log(`Total Test Time:  ${this.result?.response?.details?.runTestResult?.totalTime}`);
+    }
+  }
+}
+
+const mdResponseSorter = (i: DeployMessage, j: DeployMessage): number => {
```

Comment:
This is so close to `ResultFormatter.sortFileResponses()`.  It would be nice to use it.  We can also handle that in a refactoring/consolidation effort.